src/components/map/components/action/layer-action/modify-action/index.js

```diff
@@ -27,10 +32,43 @@ export class VlMapModifyAction extends VlMapLayerAction {
 
   _createAction(layer) {
     const options = {
-      snapping: true,
+      snapping: this.__snappingOptions,
     };
     return new VlModifyAction(layer, this._callback, options);
   }
+
+  get __snappingOptions() {
+    if (this.dataset.vlSnapping !== undefined && this.__snappingLayers.length > 0) {
+      return {
+        layer: this.__createSnappingLayer(),
+        pixelTolerance: this.dataset.vlSnappingPixelTolerance || 10,
+        node: false,
+        vertex: false,
+      };
+    } else {
+      return true;
+    }
+  }
+
+  __createSnappingLayer() {
+    const snappingLayer = new VlCompositeVectorLayer(
+        this.__snappingLayers.map((layer) => layer._layer),
+        {}
+    );
+    const firstVectorLayer = this.__snappingLayers[0];
+    snappingLayer.setStyle(firstVectorLayer.style);
+    firstVectorLayer.addEventListener(
```

Comment:
Best practice om event listeners ook weer te verwijderen

Reply:
Voorbeeld uit bvb Side sheet component:

connectedCallback() {
    this._toggle = () => this.toggle();
    this._toggleButton.addEventListener("click", this._toggle);
  }

  disconnectedCallback() {
    this._toggleButton.removeEventListener("click", this._toggle);
  }

Reply:
Ah, je bedoelt bij het destructen van de component zelf. Cava :)